Replace skew and rotate with single transform

JimBobSquarePants opened the pull request:
Relates to https://github.com/SixLabors/ImageSharp/pull/386#pullrequestreview-78815265

Both `Skew` and `Rotate` on the `Point` struct have identical signatures with identical method bodies. 

This PR simplifies this by utilizing a single `Transform` method with the same signature as the `Vector2` equivalent. (I decided not to go `ref` for consistancy).

This allows us to turn `AffineProcessor` in my ImageSharp PR into a single base class that can handle any kind of transform. I believe to developers this also provides greater consistency across `Point`, `PointF`, and `Vector2`

tocsoft commented on src/SixLabors.Core/Primitives/Point.cs:
```diff
@@ -230,30 +227,22 @@ public static PointF Transform(Point position, Matrix3x2 matrix)
+        /// <returns>The transformed <see cref="PointF"/></returns>
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        public static Point Skew(Point point, Matrix3x2 skew) => Round(Vector2.Transform(new Vector2(point.X, point.Y), skew));
+        public static Point Transform(Vector2 position, Matrix3x2 matrix) => Round(Vector2.Transform(position, matrix));
```
This method shouldn't  be public on this class... I feels wrong to have a static transform on `Point` that takes in a `Vector2`.

JimBobSquarePants commented:
I've removed the `Vector2` overload. I added it as a potential shortcut but yeah you're right. Someone can  just round the Vector result themselves should they need it.
